### ISSUE TITLE
How to run this script on Windows

### ISSUE THREAD

**Issue by ItsJustJarvis** [myspaghetti/macos-virtualbox#529]
Hey there I'm a little confused on how to actually start this script. I am fairly new to this. I have adjusted the values to fit my needs/system but when I click on the script to run it, and review the script configuration, I press enter to continue and it just closes... am I missing something?

**Comment by ItsJustJarvis** [comment]
I have no idea how to make this work.... seriously need some direction. Here is my updated script. What do I do to make this work?
[macos-guest-virtualbox.txt](https://github.com/myspaghetti/macos-virtualbox/files/7139309/macos-guest-virtualbox.txt)

**Comment by Masamune3210** [comment]
Its a shell script, hence the .sh. You have to run it from inside a shell

**Comment by ItsJustJarvis** [comment]
Can i get more of an explanation? Steps on how to run it and have it execute correctly?

**Comment by fzslm** [comment]
i believe you need cygwin. just trying it out now. i'll report back with my progress.

**Comment by Masamune3210** [comment]
Just saying, not trying to be mean, but if you are having issues googling how to use a shell script you might need to do more research before proceeding

**Comment by ItsJustJarvis** [comment]
> Just saying, not trying to be mean, but if you are having issues googling how to use a shell script you might need to do more research before proceeding

No-one seems to want to explain it in a clear manner that someone who hasn't done this before would understand. I have googled and researched and attempted to run the script all day, but nowhere has a clear answer for a solution. I have cygwin installed, I have tried multiple different ways to run the script and nothing happens. 

I need a step by step explanation is all Im saying. What do I do, I have the script file, I have updated some of the variables. What are my next steps exactly? How do I get the script to run in cygwin?

**Comment by fzslm** [comment]
i'm honestly not sure what could be going wrong for you... i just installed cygwin, installed the dependencies, and it worked first time. or, at least, it's downloading the installation files right nowm, which is further than you seem to have gotten...

do you get any terminal output all?

(fwiw: when installing dependencies, i had problems when using the latest version of `wget` that cygwin provided in its repos - i had to use one version earlier, and that fixed it for me. perhaps this could be your issue?)

**Comment by ItsJustJarvis** [comment]
> i'm honestly not sure what could be going wrong for you... i just installed cygwin, installed the dependencies, and it worked first time. or, at least, it's downloading the installation files right now, which is further than you seem to have gotten...
> 
> do you get any terminal output all? anything for us to go off?
> 
> (fwiw: when installing dependencies, i had problems when using the latest version of `wget` that cygwin provided in its repos - i had to use one version earlier, and that fixed it for me. perhaps this could be your issue?)

Can you explain the steps and what I enter into cygwin to run the script.... I am new to commands and have no clue what Im doing. I have the github repo downloaded. I edited the script in VSCode and saved my changes. What do I do now. I just need the steps in a clear and simple explaination.

**Comment by fzslm** [comment]
yeah. sure thing.

first, make sure you installed all of the needed dependencies. that's wget, coreutils, xxd, gzip, unzip. those can be installed using Cygwin's setup utility.

next, take note of where the script is stored on the Windows side of things (let's say C:\Users\name\Downloads\whatever or whatever)

open up Cygwin's terminal - this'll drop you in a bash prompt. cd into your C: (or whatever) drive, which in Cygwin is represented as /cygdrive/c

`cd /cygdrive/c`

then, cd into the folder the script is inside:

`cd Users/name/Downloads/whatever`

then, run `./macos-guest-virtualbox.sh` and you should be good to go.
